default/python/turn_events/turn_events.py

```diff
@@ -31,4 +32,35 @@ def execute_turn_events():
         if fo.create_field(field_type, x, y, size) == fo.invalid_object():
             print >> sys.stderr, "Turn events: couldn't create new field"
 
+    gsd = fo.get_galaxy_setup_data()
+    # creating monsters
+    monster_freq = fo.monster_frequency(gsd.monsterFrequency)
+    # monster freq ranges from 30 (= one monster per 30 systems) to 3 (= one monster per 3 systems)
+    # (example: low monsters and 150 Systems results in 150 / 30 * 0.001 = 0.005)
+    if monster_freq > 0 and random() < len(systems) / monster_freq * 0.001:
+        if random() < 0.7:
+            monster_type = "SM_KRILL_1"
+        else:
+            monster_type = "SM_FLOATER"
```

Comment:
If I understand correctly, this addition to the turn events is supposed to be a proof of concept. For that I think it's sufficient if we restrict the creation of new monsters to Krill Swarms initially, adding Floaters here would be an introduction of a change in mechanics, even if only a slight one.

Floaters are supposed to be spawned by Dyson Forests, not appear out of nowhere. That makes sense for Krill Swarms which originally were supposed to act as "food" for Kraken IIRC (and therefore actually need to be respawned sufficiently frequently, so this patch actually provides what is necessary to pick up that Kraken food concept), but not for other kinds of monsters, their spawning is determined by other rules/mechanics.

Changing that is a concept that should be subject to design discussion first.